pkg/controller/controller.go

```diff
@@ -59,6 +59,13 @@ type Options struct {
 	// The overall is a token bucket and the per-item is exponential.
 	RateLimiter ratelimiter.RateLimiter
 
+	// NewWorkQueue constructs the queue for this controller once the controller is ready to start.
+	// This is a func because the standard Kubernetes work queues start themselves immediately, which
+	// leads to goroutine leaks if something calls controller.New repeatedly.
+	// The NewWorkQueue func gets the controller name and the RateLimiter option (defaulted if necessary) passed in.
+	// NewWorkQueue defaults to NewRateLimitingQueueWithConfig.
+	NewWorkQueue func(controllerName string, rateLimiter ratelimiter.RateLimiter) workqueue.RateLimitingInterface
```

Comment:
```suggestion
	// NewWorkQueue constructs the queue for this controller once the controller is ready to start.
	// This is a func because the standard Kubernetes work queues start themselves immediately, which
	// leads to goroutine leaks if something calls controller.New repeatedly.
	// The NewWorkQueue func gets the controller name and the RateLimiter option (defaulted if necessary) passed in.
	// NewWorkQueue defaults to NewRateLimitingQueueWithConfig.
	NewQueue func(controllerName string, rateLimiter ratelimiter.RateLimiter) workqueue.RateLimitingInterface
```